server/src/main/java/io/druid/segment/indexing/DataSchema.java

```diff
@@ -68,23 +75,31 @@ public DataSchema(
             parser.getParseSpec()
                   .withDimensionsSpec(
                       dimensionsSpec
-                            .withDimensionExclusions(
-                                Sets.difference(dimensionExclusions,
-                                                Sets.newHashSet(dimensionsSpec.getDimensions()))
-                            )
+                          .withDimensionExclusions(
+                              Sets.difference(
+                                  dimensionExclusions,
+                                  Sets.newHashSet(dimensionsSpec.getDimensions())
+                              )
+                          )
                   )
         );
       } else {
         this.parser = parser;
       }
     } else {
+      log.warn("No parser or parseSpec has been specified");
```

Comment:
What happens with a null parseSpec? Is that a valid use-case? If not we should not allow it